sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/RenameTableExec.scala

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.execution.datasources.v2
+
+import org.apache.spark.sql.catalyst.InternalRow
+import org.apache.spark.sql.catalyst.expressions.Attribute
+import org.apache.spark.sql.connector.catalog.{Identifier, TableCatalog}
+
+/**
+ * Physical plan node for renaming a table.
+ */
+case class RenameTableExec(
```

Comment:
Are there any specific reasons for naming it as `RenameTableExec` instead of `AlterTableRenameExec` similarly to other alter table exec nodes?

Reply:
It's not a big deal, and I prefer `RenameTableExec` as it's shorter.

Reply:
`DropPartitionExec` is shorter than `AlterTableDropPartitionExec` too but we have the `AlterTable` prefix in all `ALTER TABLE` exec nodes.

Looks slightly strange that all alter table commands begin from `AlterTable` except only one.

Reply:
How many `AlterTable*` v2 commands out there? If it's a lot then let's update this one as well.

Reply:
Not a a lot but we have for now:
- AlterTableAddPartitionExec
- AlterTableRenamePartitionExec
- AlterTableDropPartitionExec
- AlterTableExec

Reply:
Then it's probably better to change them to `AddPartitionExec`, etc.

Reply:
Here is the PR https://github.com/apache/spark/pull/31584